lib/ansible/modules/cloud/amazon/elasticache_replication_group.py

```diff
@@ -0,0 +1,585 @@
+              description: Port for the endpoint of the current node
+              returned: always
+              type: int
+              sample: 6379
+    primary_endpoint:
+      description: Endpoint for the Primary node in the replication group
+      returned: always
+      type: complex
+      contains:
+        address:
+          description: Address of the replication group endpoint
+          returned: always
+          type: string
+          sample: ansible-testing-erg.qh5gss.ng.0001.use2.cache.amazonaws.com
+        port:
+          description: Port of the replication group endpoint
+          returned: always
+          type: int
+          sample: 6379
+    status:
+      description: Status of the node group
+      returned: always
+      type: string
+      sample: available
+pending_modified_values:
+  description: Values that are awaiting a reboot to complete modification
+  returned: always
+  type: complex
+  contains: {}
+replication_group_id:
+  description: ID of the Replication Group
+  returned: always
+  type: string
+  sample: ansible-testing-erg
+status:
+  description: Status of the replication group
+  returned: always
+  type: string
+  sample: available
+'''
+
+from ansible.module_utils.ec2 import boto3_conn, get_aws_connection_info, ec2_argument_spec
+from ansible.module_utils.ec2 import HAS_BOTO3, camel_dict_to_snake_dict
+from ansible.module_utils.aws.core import AnsibleAWSModule
+
+
+try:
+    import botocore
+except ImportError:
+    pass  # caught by imported HAS_BOTO3
+
+
+def check_for_rep_group(module, connection):
+
+    params = dict()
+
+    params['ReplicationGroupId'] = module.params['name']
+
+    try:
+        result = connection.describe_replication_groups(**params)
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        if e.response['Error']['Code'] == 'ReplicationGroupNotFoundFault':
+            return None
+        else:
+            module.fail_json_aws(e, msg="Couldn't search for replication group %s" % module.params['name'])
+
+    return result['ReplicationGroups'][0]
+
+
+def create_rep_group(module, connection):
+
+    params = dict()
+
+    params['ReplicationGroupId'] = module.params.get('name')
+    params['ReplicationGroupDescription'] = module.params.get('description')
+    params['PrimaryClusterId'] = module.params.get('primary_cluster_id')
+    params['AutomaticFailoverEnabled'] = module.params.get('failover_enabled')
+    params['NumCacheClusters'] = module.params.get('num_cache_clusters')
+    params['PreferredCacheClusterAZs'] = module.params.get('preferred_cache_cluster_azs')
+    params['CacheNodeType'] = module.params.get('node_type')
+    params['Engine'] = module.params.get('engine')
+    params['EngineVersion'] = module.params.get('cache_engine_version')
+    params['CacheParameterGroupName'] = module.params.get('cache_parameter_group')
+    params['CacheSubnetGroupName'] = module.params.get('cache_subnet_group_name')
```

Comment:
This needs to be `CacheSubnetGroupNames`.  And the associated module parameter should be set to a 'list' type.